pkg/apis/intevents/v1beta1/topic_conversion_test.go

```diff
@@ -18,11 +18,43 @@ package v1beta1
 
 import (
 	"context"
+	"net/url"
 	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1 "github.com/google/knative-gcp/pkg/apis/intevents/v1"
+	gcptesting "github.com/google/knative-gcp/pkg/testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/apis/duck/v1beta1"
+)
+
+// These variables are used to create a 'complete' version of Topic where every field is filled in.
+var (
+	// completeTopic is a Topic with every filled in, except TypeMeta. TypeMeta is excluded because
+	// conversions do not convert it and this variable was created to test conversions.
+	completeTopic = &Topic{
+		ObjectMeta: gcptesting.CompleteObjectMeta,
+		Spec: TopicSpec{
+			IdentitySpec:      gcptesting.CompleteV1beta1IdentitySpec,
+			Secret:            gcptesting.CompleteSecret,
+			Project:           "project",
+			Topic:             "topic",
+			PropagationPolicy: TopicPolicyCreateDelete,
+			EnablePublisher:   &trueVal,
+		},
+		Status: TopicStatus{
+			IdentityStatus: gcptesting.CompleteV1beta1IdentityStatus,
+			AddressStatus:  gcptesting.CompleteV1beta1AddressStatus,
+			ProjectID:      "projectID",
+			TopicID:        "topicID",
+		},
+	}
 )
 
 func TestTopicConversionBadType(t *testing.T) {
-	good, bad := &Topic{}, &Topic{}
+	good, bad := &Topic{}, &PullSubscription{}
```

Comment:
Why it's `&Topic{}` vs `&PullSubscription{}`?
The logic is correct, just the code style is not consistent with other resources.

Reply:
I'm actually keep the code style consistent with other resources here, as they are all using explicit different types. 
Check: https://github.com/google/knative-gcp/blob/master/pkg/apis/intevents/v1alpha1/pullsubscription_conversion_test.go#L171 and https://github.com/google/knative-gcp/blob/master/pkg/apis/events/v1alpha1/cloudschedulersource_conversion_test.go#L54

Reply:
Ok. I think the codebase uses different code style. 
Check https://github.com/google/knative-gcp/blob/e76eb0926f07aa97d59cfb66742f28361cfdc831/pkg/apis/intevents/v1beta1/pullsubscription_conversion_test.go#L25.
We can clean it in follow up PRs.